# AC2021_NANOPORE_PCR10  107_81_45_21Q_MIX (114 barcode-kit)

## Basecalling

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/
./submit_chronos_gianluca guppy-gianluca-job-basecalling.json
```

## STRIQUE

### STRIQUE INDEX

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/
condor_submit -name ettore condor-striqueIndex.job
```

### alignment reads "both_ends" with PCR fragment

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/

for n in $(seq 19 21) ; do ./submit_chronos_gianluca guppy-gianluca-job-align${n}.json ; done
```

### numero di reads allineate con fasta 107Q flippato (senza NEO)

```bash
for n in $(seq 19 21) ; do cat barcode${n}/alignment_summary.txt | grep 107Q | wc -l ; done
```

|barcode |reads number|
|------|--------------|
| barcode19 | 723727  |
| barcode20 | 290547  |
| barcode21 | 263193  |
| barcode22 | 741319  |
| barcode23 | 586056  |
| barcode24 | 547249  |

### 3.2 - STRique count Analysis

#### 3.2.1 concat all SAM file after alignment

```bash
for n in $(seq 19 21) ; do bash /lustre/home/enza/cattaneo/src/concatSAM_split4STRique_PCR10.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/alignment/barcode${n} 107Q ; done
```

```bash
for c in $(seq 19 21) ; do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10 ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/alignment/barcode${c}/${f2}" -a "fofn=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/barcode${c}/AC_barcode${c}_MIX.${f2}.qs10.150_770bp_CAG.tsv" -a "file=${f2}" -a "barcode=${c}" /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/condor-striqueCount_exp.job ; done ; done; done | less -S
```

### 4.2 Merge splitted STRique output

```bash
for n in $(seq 19 21) ;  do cp barcode${n}/AC_barcode${n}_MIX.split_fq_aa.qs10.150_770bp_CAG.tsv barcode${n}/AC_barcode${n}_MIX.qs10.150_770bp_CAG.tsv ; done

for n in $(seq 19 21) ; do ls barcode${n}/AC_barcode${n}_MIX.split_fq_* | grep -v aa | while read line ; do cat $line | tail -n+2 >> barcode${n}/AC_barcode${n}_MIX.qs10.150_770bp_CAG.tsv ; done ; done
```

## STRAGLR

Update 2022-02-03 for comparisons with run: mix1-3_81Q_prolif_c8_c30_c47

### FASTQ2FASTA

```bash
mkdir /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/fastq2fasta

for n in $(seq 19 21); do zcat /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/basecalling/pass/barcode${n}/*.fastq.gz | awk '{if(NR%4==1) {printf(">%s\n",substr($0,2));} else if(NR%4==2) print;}' > /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/fastq2fasta/barcode${n}_allreads.fasta ; done
```

### BLASTN all CAG sizes (21Q, 45Q, 81Q, 107Q) for barcode19-21

```bash
source ~/.bashrc

mkdir /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX

for c in 21Q 45Q 81Q 107Q ; do for n in $(seq 19 21); do blastn -task blastn-short -query /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/fastq2fasta/barcode${n}_allreads.fasta -subject /lustrehome/gianluca/straglr/barcode_fasta/barcode${c}.fa -outfmt 6 > /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/barcode${n}_blastn_${c}.txt ; done ; done 
```

### WHOLE GENOME alignment

> Barcode 19-21 mix

```bash
for n in $(seq 19 21); do cp guppy-gianluca-job-WG-align19.json guppy-gianluca-job-WG-align${n}.json ; done

for n in $(seq 19 21); do sed -i -e "s/barcode19/barcode${n}/g" guppy-gianluca-job-WG-align${n}.json ; done

for n in $(seq 19 21); do ./submit_chronos_gianluca guppy-gianluca-job-WG-align${n}.json ; done
```

### *straglr*, for CAG count

CL Merge 100 bam:

```bash
for n in $(seq 19 21) ; do bash bash-merge_100_BAM.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/alignment/barcode${n}_WG ; done
```

CL Job submission:

```bash
mkdir /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX

cd /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/alignment/

for n in $(seq 19 21) ; do ls barcode${n}_WG/*.multi.bam | while IFS="$(printf '/')" read -r f1 f2 ; do echo condor_submit -name ettore \
-a "out_dir=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/" \
-a "bam=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/107_81_45_21Q_MIX/20211130_1922_MN37986_FAR65790_29871ba8/analysis/alignment/barcode${n}_WG/${f2}" \
-a "ref=/lustre/home/enza/cattaneo/data/Methylation-RUES2-FLG/RUES2-20CAG/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna" \
-a "out_prefix=barcode${n}_${f2}" \
-a "bed=/lustrehome/gianluca/test_straglr/CAG.bed" \
-a "file=${f2}" \
-a "barcode=barcode${n}" \
/lustrehome/gianluca/straglr/jobs/condor-straglr_exp.job ; done ; done | less -S
```
