# AC2021_NANOPORE_PCR10 IC046 D0(3)-D45+

## Basecalling

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10
submit_chronos_gianluca guppy-gianluca-job-diff2-basecalling.json
```

## STRIQUE

### STRIQUE INDEX

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10
condor_submit -name ettore condor-striqueIndex.job
```

alignment reads "both_ends" with PCR fragment

```bash
for c in $(seq 13 20); do submit_chronos_gianluca guppy-gianluca-job-diff2-align${n}.json ; done
```

numero di reads allineate con fasta 107Q flippato (senza NEO)

```bash
-bash-4.2$ for n in $(seq 13 20) ; do cat barcode$n/alignment_summary.txt | grep 107Q | wc -l; done
258491
285799
120990
212122
227660
173442
205078
198996
```

### 3.2 - STRique count Analysis

#### 3.2.1 concat all SAM file after alignment

```bash
for n in 13 14 15 16 17 18 19 20; do bash /lustre/home/enza/cattaneo/src/concatSAM_split4STRique_PCR10.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${n} 107Q ; done
```

#### 3.2.2 STRique job submission

```bash
for c in 13 14 15 16 17 18 19 20; do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10 ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${c}/${f2}" -a "fofn=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/barcode${c}/AC_barcode${c}_diff2.${f2}.qs10.150_770bp_CAG.tsv" -a "file=${f2}" -a "barcode=${c}"  ~/strique/jobs/AC2021_NANOPORE_PCR10/condor-striqueCount_exp.job ; done ; done; done | less -S

for c in 13 ; do for i in c d e f g h ; do for g in {a..z} ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${c}/split_fq_${i}${g}" -a "fofn=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/barcode${c}/AC_barcode${c}_diff2.split_fq_${i}${g}.qs10.150_770bp_CAG.tsv" -a "file=split_fq_${i}${g}" -a "barcode=${c}"  ~/strique/jobs/AC2021_NANOPORE_PCR10/condor-striqueCount_exp.job ; done ; done; done | less -S

for c in 13 ; do for i in a ; do for g in {l..o} ; do for n in 10 ; do echo qsub -q testqueue -l nodes=1:ppn=40 -o /lustrehome/gianluca/junk/cattaneo/striqueCount_barcode${c}.split_fq_${i}${g}.qscore$n.out -e /lustrehome/gianluca/junk/cattaneo/striqueCount_barcode${c}.split_fq_${i}${g}.qscore$n.err -v sam="/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${c}/split_fq_${i}${g}",fofn="/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/fast5/reads.fofn",config="/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv",output="/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/barcode${c}/AC_barcode${c}_diff2.split_fq_${i}${g}.qs10.150_770bp_CAG.tsv" -N barcode${c}.split_fq_${i}${g}.qs$n.count /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/pbs-striqueCount.job ; done ; done; done; done | less -S

#### CL DA CONSIDERARE:

for c in 13 17 14 18 15 19 16 20; do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10; do echo qsub -q testqueue -l nodes=1:ppn=40 -o /lustrehome/gianluca/junk/cattaneo/striqueCount_barcode${c}.${f2}.qscore$n.out -e /lustrehome/gianluca/junk/cattaneo/striqueCount_barcode${c}.${f2}.qscore$n.err -v sam="/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${c}/${f2}",fofn="/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/fast5/reads.fofn",config="/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv",output="/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/barcode${c}/AC_barcode${c}_diff2.${f2}.qs10.150_770bp_CAG.tsv" -N barcode${c}.${f2}.qs$n.count /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/pbs-striqueCount.job ; done ; done; done| less -S
```

### 4.2 Merge splitted STRique output

```bash
for n in $(seq 13 20) ;  do cp barcode${n}/AC_barcode${n}_diff2.split_fq_aa.qs10.150_770bp_CAG.tsv barcode${n}/AC_barcode${n}_diff2.qs10.150_770bp_CAG.tsv ; done

for n in $(seq 13 20) ; do ls barcode${n}/AC_barcode${n}_diff2.split_fq_* | grep -v aa | while read line ; do cat $line | tail -n+2 >> barcode${n}/AC_barcode${n}_diff2.qs10.150_770bp_CAG.tsv ; done ; done
```

### Multiple Peaks Detection (MPD)

~ re-run old experiment for improve the peaks detection

#### using multiple peaks detection for spike-in 45q_105q (PASTECS : -log2(pvalue) > 5 ; MFT : window_len = 20,25,30 ; Z-SCORE : lag = 15,20,25,30, treshold = 4.5, influence = 0.1)

#### QSCORE 10 alignment vanilla & m13_barcode

##### qscore10

```bash
for p in 13 17 ; do for c in long; do Rscript /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/peaksDetectComp_2021_PCR10.R -p AC2021_NANOPORE_barcode${p} -e '105' -l $c -m '15,16,17,18,19,20' -z '15,20,25,30' -s /Users/gianlucadamaggio/projects/cattaneo/striqueOutput/AC2021_NANOPORE_PCR10/barcode${p}/AC_barcode${p}_diff2.qs10.150_770bp_CAG.tsv -a /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/alignment/barcode${p}/alignment_summary.txt  -f /Users/gianlucadamaggio/Desktop/multiplePeaksDetector/multiplePeaksDetector_AC2021_NANOPORE_PCR10/barcode${p}/ ; done ;done
```

##### qscore7

```bash
for p in 13 17 ; do for c in long; do Rscript ~/projects/cattaneo/docs/AC2021_NANOPORE_PCR7/peaksDetectComp_107Q_p83-92.R -p AC2021_NANOPORE_barcode${p} -e '105' -l $c -m '15,16,17,18,19,20' -z '15,20,25,30' -s /Users/gianlucadamaggio/projects/cattaneo/striqueOutput/AC2021_NANOPORE_PCR7_DIFF_IC046/barcode${p}/AC_barcode${p}_both_ends.qs7.150_900bp_CAG.tsv -a /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR7_DIFF_IC046/alignment/barcode${p}/qscore7/alignment_summary.txt  -f /Users/gianlucadamaggio/Desktop/multiplePeaksDetector/multiplePeaksDetector_AC2021_NANOPORE_PCR7_DIFF_IC046/barcode${p}/qscore7/ ; done ;done
```

## STRAGLR

### WHOLE GENOME alignment

Add to json file for alignment vs the whole-genome.

```json
"constraints": [["hostname", "LIKE", "hpc-gpu-1-[2,4]-[1-2].recas.ba.infn.it"]],
```

107Q low-density

```bash
for n in $(seq 13 20) ; do cp guppy-gianluca-job-WG-align13.json guppy-gianluca-job-WG-align${n}.json ; done

for n in $(seq 13 20) ; do sed -i -e "s/barcode13/barcode${n}/g" guppy-gianluca-job-WG-align${n}.json ; done

for n in $(seq 13 20) ; do ./submit_chronos_gianluca guppy-gianluca-job-WG-align${n}.json ; done
```

### FASTQ2FASTA

```bash
mkdir /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/fastq2fasta

for n in $(seq 13 20) ; do zcat /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/basecalling/pass/barcode${n}/*.fastq.gz | awk '{if(NR%4==1) {printf(">%s\n",substr($0,2));} else if(NR%4==2) print;}' > /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/fastq2fasta/barcode${n}_allreads.fasta ; done
```

### BLASTN 107Qbarcode

```bash
source ~/.bashrc

mkdir /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20) ; do blastn -task blastn-short -query /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/fastq2fasta/barcode${n}_allreads.fasta -subject /lustrehome/gianluca/straglr/barcode_fasta/barcode107Q.fa -outfmt 6 > /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/barcode${n}_blastn_107Q.txt ; done
```

### *straglr*, for CAG count

CL Merge 100 bam:

```bash
for n in $(seq 13 20) ; do bash bash-merge_100_BAM.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${n}_WG ; done
```

CL Job submission:

```bash
mkdir /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

cd /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/

for n in $(seq 13 20) ; do ls barcode${n}_WG/*.multi.bam | while IFS="$(printf '/')" read -r f1 f2 ; do echo condor_submit -name ettore \
-a "out_dir=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/" \
-a "bam=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${n}_WG/${f2}" \
-a "ref=/lustre/home/enza/cattaneo/data/Methylation-RUES2-FLG/RUES2-20CAG/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna" \
-a "out_prefix=barcode${n}_${f2}" \
-a "bed=/lustrehome/gianluca/test_straglr/CAG.bed" \
-a "file=${f2}" \
-a "barcode=barcode${n}" \
/lustrehome/gianluca/straglr/jobs/condor-straglr_exp.job ; done ; done | less -S
```

## Merge multiple *straglr* output

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20) ; do ls barcode${n}_split*.tsv | grep aa | while read file ; do cat ${file} > barcode${n}.tsv; done ; done

for n in $(seq 13 20) ; do ls barcode${n}_split*.tsv | grep -v aa | while read file ; do cat ${file} | grep -v "#" >> barcode${n}.tsv; done ; done

#cancel all .ok files created for job-control
rm *.ok
```

## Create folder to save on local computer with blastn+straglr+alignment_summary

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20) ; do mkdir barcode${n} ; done

ALIGNMENT=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment

BLASTN=/lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

STRAGLR=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20) ; do cp ${ALIGNMENT}/barcode${n}_WG/alignment_summary.txt ${STRAGLR}/barcode${n}/ &&  cp ${BLASTN}/barcode${n}* ${STRAGLR}/barcode${n}/ &&  cp ${STRAGLR}/barcode${n}.tsv ${STRAGLR}/barcode${n}/ ; done 
```

## Rsync to local computer

```bash
mkdir /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/

for n in $(seq 13 20) ; do rsync -avh --progress gianluca@ui02.recas.ba.infn.it:/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/barcode${n} /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/ ; done
```

## Re-do Straglr - version CAG_v4.bed

```bash
mkdir /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

cd /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/

for n in $(seq 13 20) ; do ls barcode${n}_WG/*.multi.bam | while IFS="$(printf '/')" read -r f1 f2 ; do echo condor_submit -name ettore \
-a "out_dir=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/" \
-a "bam=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/20210722_1913_MN29119_FAP20649_23b1dbb6/analysis/alignment/barcode${n}_WG/${f2}" \
-a "ref=/lustre/home/enza/cattaneo/data/Methylation-RUES2-FLG/RUES2-20CAG/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna" \
-a "out_prefix=barcode${n}_v4_${f2}" \
-a "bed=/lustrehome/gianluca/test_straglr/CAG_v4.bed" \
-a "file=${f2}" \
-a "barcode=barcode${n}" \
/lustrehome/gianluca/straglr/jobs/condor-straglr_exp.job ; done ; done | less -S
```

### Merge multiple *straglr* output - version CAG_v4.bed

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20); do ls barcode${n}_v4_split*.tsv | grep aa | while read file ; do cat ${file} > barcode${n}_v4.tsv; done ; done

for n in $(seq 13 20); do ls barcode${n}_v4_split*.tsv | grep -v aa | while read file ; do cat ${file} | grep -v "#" >> barcode${n}_v4.tsv; done ; done

#cancel all .ok files created for job-control
rm *.ok
```

### Create folder to save on local computer with blastn+straglr+alignment_summary - version CAG_v4.bed

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20) ; do mkdir barcode${n}_v4 ; done

STRAGLR=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL

for n in $(seq 13 20); do cp ${STRAGLR}/barcode${n}_v4.tsv ${STRAGLR}/barcode${n}_v4/ ; done 
```

### Rsync to local computer - version CAG_v4.bed

```bash
mkdir /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/

for n in $(seq 13 20) ; do rsync -avh --progress gianluca@ui02.recas.ba.infn.it:/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/barcode${n}_v4 /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/D0_D45_PAN_NEURAL/ ; done
```

-->