# AC2021_NANOPORE_PCR10 D0_D40_low_IC050


#### Basecalling
```
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/
./submit_chronos_gianluca guppy-gianluca-job-lowDiff-basecalling.json
```

#### STRIQUE INDEX
```
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/
condor_submit -name ettore condor-striqueIndex.job
```

#### alignment reads "both_ends" with PCR fragment
```
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/
for n in $(seq 17 24); do ./submit_chronos_gianluca guppy-gianluca-job-diffLow-align${n}.json ; done
```
#### numero di reads allineate con fasta 107Q flippato (senza NEO)
```
for n in $(seq 17 24) ; do cat barcode$n/alignment_summary.txt | grep 107Q | wc -l ; done
```
|barcode |reads number|
|------|--------------|
|barcode13 | 633386 |
|barcode14 | 569406 |
|barcode15 | 465734 |
|barcode16 | 678476 |
|barcode17 | 457240 |
|barcode18 | 661200 |
|barcode19 | 623623 |
|barcode20 | 484804 |

#### 3.2 - STRique count Analysis
##### 3.2.1 concat all SAM file after alignment
```
for n in $(seq 17 24); do bash /lustre/home/enza/cattaneo/src/concatSAM_split4STRique_PCR10.sh /lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/20210903_1730_MN29119_FAQ69443_40596b12/analysis/alignment/barcode${n} 107Q ; done
```
export HDF5_PLUGIN_PATH=/lustrehome/gianluca/src/ont_fast5_api/ont_fast5_api/vbz_plugin/

```
for c in $(seq 17 24); do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10 ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/20210903_1730_MN29119_FAQ69443_40596b12/analysis/alignment/barcode${c}/${f2}" -a "fofn=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/20210903_1730_MN29119_FAQ69443_40596b12/fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_150_770bp_CAG.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/barcode${c}/AC_barcode${c}_diffLow.${f2}.qs10.150_770bp_CAG.tsv" -a "file=${f2}" -a "barcode=${c}" /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/condor-striqueCount_exp.job ; done ; done; done | wc -l
```

#### 4.2 Merge splitted STRique output
```
for n in $(seq 17 24) ;  do cp barcode${n}/AC_barcode${n}_diffLow.split_fq_aa.qs10.150_770bp_CAG.tsv barcode${n}/AC_barcode${n}_diffLow.qs10.150_770bp_CAG.tsv ; done

for n in $(seq 17 24) ; do ls barcode${n}/AC_barcode${n}_diffLow.split_fq_* | grep -v aa | while read line ; do cat $line | tail -n+2 >> barcode${n}/AC_barcode${n}_diffLow.qs10.150_770bp_CAG.tsv ; done ; done
```

##### Rhmm

```
for n in 17;  do echo qsub -q testqueue -l nodes=1:ppn=40 -o /lustrehome/gianluca/junk/cattaneo/rhmm_barcode${n}.out -e /lustrehome/gianluca/junk/cattaneo/rhmm_barcode${n}.err -v id="${n}" -N rhmm_barcode${n} ~/strique/jobs/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/pbs-lowDensity_barcode.sh ; done | less -S

```

# Multiple Peaks Detection (MPD)
~ re-run old experiment for improve the peaks detection

#### using multiple peaks detection for spike-in 45q_105q (PASTECS : -log2(pvalue) > 5 ; MFT : window_len = 20,25,30 ; Z-SCORE : lag = 15,20,25,30, treshold = 4.5, influence = 0.1)
## QSCORE 10 alignment vanilla & m13_barcode

## qscore10
```
for p in $(seq 17 24) ; do for c in long; do Rscript /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/peaksDetectComp_2021_PCR10.R -p AC2021_NANOPORE_barcode${p} -e '105' -l $c -m '15,16,17,18,19,20' -z '15,20,25,30' -s /Users/gianlucadamaggio/projects/cattaneo/striqueOutput/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/barcode${p}/AC_barcode${p}_diffLow.qs10.150_770bp_CAG.tsv -a /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/alignment/D0_D40_low_IC050/barcode${p}/alignment_summary.txt  -f /Users/gianlucadamaggio/Desktop/multiplePeaksDetector/multiplePeaksDetector_AC2021_NANOPORE_PCR10/D0_D40_low_IC050/barcode${p}/ ; done ;done

for p in 20 ; do for c in long; do Rscript /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/peaksDetectComp_2021_PCR10.R -p AC2021_NANOPORE_barcode${p} -e '105' -l $c -m '15,16,17,18,19,20' -z '10,12,15' -s /Users/gianlucadamaggio/projects/cattaneo/striqueOutput/AC2021_NANOPORE_PCR10/D0_D40_low_IC050/barcode${p}/AC_barcode${p}_diffLow.qs10.150_770bp_CAG.tsv -a /Users/gianlucadamaggio/projects/cattaneo/docs/AC2021_NANOPORE_PCR10/alignment/D0_D40_low_IC050/barcode${p}/alignment_summary.txt  -f /Users/gianlucadamaggio/Desktop/multiplePeaksDetector/multiplePeaksDetector_AC2021_NANOPORE_PCR10/D0_D40_low_IC050/barcode${p}/ ; done ;done
```
