# AC2021_NANOPORE_PCR10 21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120 ( 104, 114 barcode-kit) 1-15

## Basecalling

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/
./submit_chronos_gianluca guppy-gianluca-job-basecalling.json
```

## STRIQUE INDEX

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/
condor_submit -name ettore condor-striqueIndex.job
```

## alignment reads "both_ends" with PCR fragment

```bash
cd /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/
```

Barcode 01-05 21Q

```bash
for n in 02 03 04 05; do cp guppy-gianluca-job-align01.json guppy-gianluca-job-align${n}.json ; done

for n in 01 02 03 04 05 ; do sed -i -e "s/barcode01/barcode${n}/g" guppy-gianluca-job-align${n}.json ; done

for n in 01 02 03 04 05 ; do ./submit_chronos_gianluca guppy-gianluca-job-align${n}.json ; done
```

Barcode 06-15 45Q

```bash
for n in 06 07 08 09 10 11 12 13 14 15; do cp guppy-gianluca-job-align01.json guppy-gianluca-job-align${n}.json ; done

for n in 06 07 08 09 10 11 12 13 14 15; do sed -i -e "s/barcode01/barcode${n}/g" guppy-gianluca-job-align${n}.json ; done
for n in 06 07 08 09 10 11 12 13 14 15; do sed -i -e "s/AC2021_NANOPORE_PCR10_21Q_FLP.fa/AC2021_NANOPORE_PCR10_45Q_FLP.fa/g" guppy-gianluca-job-align${n}.json ; done

for n in 06 07 08 09 10 11 12 13 14 15; do ./submit_chronos_gianluca guppy-gianluca-job-align${n}.json ; done
```

## WHOLE GENOME alignment

Barcode 01-05 21Q

```bash
for n in 02 03 04 05; do cp guppy-gianluca-job-WG-align01.json guppy-gianluca-job-WG-align${n}.json ; done

for n in 01 02 03 04 05 ; do sed -i -e "s/barcode01/barcode${n}/g" guppy-gianluca-job-WG-align${n}.json ; done

for n in 01 02 03 04 05 ; do ./submit_chronos_gianluca guppy-gianluca-job-WG-align${n}.json ; done
```

```bash
for n in 06 07 08 09 10 11 12 13 14 15; do cp guppy-gianluca-job-WG-align01.json guppy-gianluca-job-WG-align${n}.json ; done

for n in 06 07 08 09 10 11 12 13 14 15; do sed -i -e "s/barcode01/barcode${n}/g" guppy-gianluca-job-WG-align${n}.json ; done

for n in 06 07 08 09 10 11 12 13 14 15; do ./submit_chronos_gianluca guppy-gianluca-job-WG-align${n}.json ; done
```

## straglr, for count CAG

merge bam from alignment

```bash
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 ; do echo bash /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/bash-merge_multiBAM.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n}_WG ; done
```

```bash
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15  ; do echo qsub -q testqueue -l nodes=1:ppn=10 \
    -o /lustrehome/gianluca/junk/cattaneo/straglr_barcode${n}.out \
    -e /lustrehome/gianluca/junk/cattaneo/straglr_barcode${n}.err \
    -v out_dir="/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120",bam="/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n}_WG/all_reads.sorted.bam",ref="/lustre/home/enza/cattaneo/data/Methylation-RUES2-FLG/RUES2-20CAG/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna",out_prefix="barcode${n}",bed="/lustrehome/gianluca/test_straglr/CAG.bed" \
    -N straglr_count_barcode${n} \
    /lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/pbs-straglr.job ; done | less -S
```

## BLASTN

### barcode21Q

Conversione fastq in fasta

```bash
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do zcat /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/basecalling/pass/barcode${n}/*.fastq.gz | awk '{if(NR%4==1) {printf(">%s\n",substr($0,2));} else if(NR%4==2) print;}' > /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/fastq2fasta/barcode${n}_allreads.fasta ; done
```
<!-- bgzip: -->
<!-- for n in 01 ; do zcat /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/basecalling/pass/barcode${n}/*.fastq.gz | awk '{if(NR%4==1) {printf(">%s\n",substr($0,2));} else if(NR%4==2) print;}' | /lustrehome/gianluca/.conda/envs/repeathmmenv/bin/bgzip > /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/fastq2fasta/barcode${n}_allreads.fasta.gz ; done -->

BLASTN su barcode21Q

```bash
source ~/.bashrc

for n in 01 02 03 04 05 ; do blastn -task blastn-short -query /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/fastq2fasta/barcode${n}_allreads.fasta -subject /lustrehome/gianluca/straglr/barcode_fasta/barcode21Q.fa -outfmt 6 > /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${n}_blastn_21Q.txt ; done
```

BLASTN su barcode21Q

```bash
source ~/.bashrc

for n in 06 07 08 09 10 11 12 13 14 15 ; do blastn -task blastn-short -query /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/fastq2fasta/barcode${n}_allreads.fasta -subject /lustrehome/gianluca/straglr/barcode_fasta/barcode45Q.fa -outfmt 6 > /lustrehome/gianluca/straglr/blastn/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${n}_blastn_45Q.txt ; done
```

STRAGLR analysis

```bash
cd /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120

Rscript straglr_plots.R
```

## numero di reads allineate con fasta 107Q flippato (senza NEO)

```bash
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 ; do cat barcode${n}/alignment_summary.txt | grep AC2021_NANOPORE_PCR10_45Q_FLP | wc -l ; done
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 ; do cat barcode${n}/alignment_summary.txt | grep AC2021_NANOPORE_PCR10_21Q_FLP | wc -l ; done
```

| barcode   | n reads|
|___________|________|
|     fasta 21Q      |
| barcode01 | 35553  |
| barcode02 | 81474  |
| barcode03 | 56648  |
| barcode04 | 106651 |
| barcode05 | 116540 |
|      fasta 45Q     |
| barcode06 | 151883 |
| barcode07 | 115960 |
| barcode08 | 81676  |
| barcode09 | 107826 |
| barcode10 | 96395  |
| barcode11 | 47137  |
| barcode12 | 69636  |
| barcode13 | 137205 |
| barcode14 | 52064  |
| barcode15 | 84823  |

## 3.2 - STRique count Analysis

### 3.2.1 concat all SAM file after alignment

```bash
cd /lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/
mkdir /lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 ; do mkdir /lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${n} ; done
```

```bash
for n in 01 02 03 04 05 ; do bash /lustre/home/enza/cattaneo/src/concatSAM_split4STRique_PCR10.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n} AC2021_NANOPORE_PCR10_21Q_FLP ; done

for n in 06 07 08 09 10 11 12 13 14 15 ; do bash /lustre/home/enza/cattaneo/src/concatSAM_split4STRique_PCR10.sh /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n} AC2021_NANOPORE_PCR10_45Q_FLP ; done
```

21Q fasta, barcode01-05

```bash
for c in 01 02 03 04 05 ; do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10 ; do echo condor_submit -name ettore \
-a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${c}/${f2}" \
-a "fofn=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/fast5/reads.fofn" \
-a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_21Q_150_530bp_CAG.tsv" \
-a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${c}/AC_barcode${c}_21Q_plurip.${f2}.qs10.150_530bp_CAG.tsv" \
-a "file=${f2}" \
-a "barcode=${c}" \
/lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/condor-striqueCount_exp.job \
; done ; done; done | less -S
```

45Q fasta, barcode06-15

```bash
for c in 06 07 08 09 10 11 12 13 14 15 ; do ls barcode${c}/split_fq_* | while IFS="$(printf '/')" read -r f1 f2 ; do for n in 10 ; do echo condor_submit -name ettore \
-a "sam=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${c}/${f2}" \
-a "fofn=/lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/fast5/reads.fofn" \
-a "config=/lustre/home/enza/cattaneo/config_file/AC2021_NANOPORE_PCR10/HTT_config_AC2021_NANOPORE_PCR10_45Q_150_600bp_CAG.tsv" \
-a "output=/lustrehome/gianluca/strique/striqueOutput/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${c}/AC_barcode${c}_45Q_plurip.${f2}.qs10.150_600bp_CAG.tsv" \
-a "file=${f2}" \
-a "barcode=${c}" \
/lustrehome/gianluca/strique/jobs/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/condor-striqueCount_exp.job \
; done ; done; done | less -S
```

### 4.2 Merge splitted STRique output

21Q barcode 1-5

```bash
for n in 01 02 03 04 05 ; do cp barcode${n}/AC_barcode${n}_21Q_plurip.split_fq_aa.qs10.150_530bp_CAG.tsv barcode${n}/AC_barcode${n}.qs10.150_530bp_CAG.tsv ; done

for n in 01 02 03 04 05 ; do ls barcode${n}/AC_barcode${n}_21Q_plurip.split_fq_* | grep -v aa | while read line ; do cat $line | tail -n+2 >> barcode${n}/AC_barcode${n}.qs10.150_530bp_CAG.tsv ; done ; done
```

45Q barcode 6-15

```bash
for n in 06 07 08 09 10 11 12 13 14 15 ; do cp barcode${n}/AC_barcode${n}_45Q_plurip.split_fq_aa.qs10.150_600bp_CAG.tsv barcode${n}/AC_barcode${n}.qs10.150_600bp_CAG.tsv ; done

for n in 06 07 08 09 10 11 12 13 14 15 ; do ls barcode${n}/AC_barcode${n}_45Q_plurip.split_fq_* | grep -v aa | while read line ; do cat $line | tail -n+2 >> barcode${n}/AC_barcode${n}.qs10.150_600bp_CAG.tsv ; done ; done
```

## Re-do Straglr - version CAG_v4.bed

CL Merge 100 bam:

```bash
for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do bash ~/straglr/jobs/bash-merge_100_BAM.sh /lustre/home/enza/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n}_WG ; done
```

CL Job submission:

```bash
mkdir /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120

cd /lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 ; do ls barcode${n}_WG/*.multi.bam | while IFS="$(printf '/')" read -r f1 f2 ; do echo condor_submit -name ettore \
-a "out_dir=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/" \
-a "bam=/lustrehome/gianluca/cattaneo/data/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/20211221_1720_MN37986_FAR73849_14cd4672/analysis/alignment/barcode${n}_WG/${f2}" \
-a "ref=/lustre/home/enza/cattaneo/data/Methylation-RUES2-FLG/RUES2-20CAG/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna" \
-a "out_prefix=barcode${n}_v4_${f2}" \
-a "bed=/lustrehome/gianluca/test_straglr/CAG_v4.bed" \
-a "file=${f2}" \
-a "barcode=barcode${n}" \
/lustrehome/gianluca/straglr/jobs/condor-straglr_exp.job ; done ; done | less -S
```

### Merge multiple *straglr* output - version CAG_v4.bed

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do ls barcode${n}_v4_split*.tsv | grep aa | while read file ; do cat ${file} > barcode${n}_v4.tsv; done ; done

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do ls barcode${n}_v4_split*.tsv | grep -v aa | while read file ; do cat ${file} | grep -v "#" >> barcode${n}_v4.tsv; done ; done

#cancel all .ok files created for job-control
rm *.ok
```

### Create folder to save on local computer with blastn+straglr+alignment_summary - version CAG_v4.bed

```bash
cd /lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do mkdir barcode${n}_v4 ; done

STRAGLR=/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do cp ${STRAGLR}/barcode${n}_v4.tsv ${STRAGLR}/barcode${n}_v4/ ; done 
```

### Rsync to local computer - version CAG_v4.bed

```bash
mkdir /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/

for n in 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15; do rsync -avh --progress gianluca@ui02.recas.ba.infn.it:/lustrehome/gianluca/straglr/output/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/barcode${n}_v4 /Users/gianlucadamaggio/projects/cattaneo/straglr/AC2021_NANOPORE_PCR10/21Q_45Q_C6_12_51_PLURIP_D0_30_60_90_120/ ; done
```
-->